rust/stracciatella_c_api/src/c/misc.rs

```diff
@@ -232,28 +219,59 @@ mod tests {
 
     #[test]
     fn check_if_relative_path_exists() {
+        // The case sensitivity of the filesystem is always unknown.
+        // Even parts of the path can have different case sensitivity.
+        // Make sure the expected result never depends on the case sensitivity of the filesystem!
+        //
+        // Different representations of the umlaut ö in utf-8:
+        // "utf8-gedöns" can be "utf8-ged\u{00F6}ns" or "utf8-gedo\u{0308}ns"
+        // "utf8-GEDÖNS" can be "utf8-GED\u{00D6}NS" or "utf8-GEDO\u{0308}NS"
+
         let temp_dir = TempDir::new().unwrap();
         fs::create_dir_all(temp_dir.path().join("foo/bar")).unwrap();
+        fs::create_dir_all(temp_dir.path().join("with space/inner")).unwrap();
+        fs::create_dir_all(temp_dir.path().join("utf8-ged\u{00F6}ns/inner")).unwrap();
 
         macro_rules! t {
-            ($path:expr, $caseless:expr, $expected:expr) => {
-                let base = CString::new(temp_dir.path().to_str().unwrap()).unwrap();
+            ($base: expr, $path:expr, $caseless:expr, $expected:expr) => {
+                let base = CString::new($base.to_str().unwrap()).unwrap();
                 let path = CString::new($path).unwrap();
                 assert_eq!(
                     super::checkIfRelativePathExists(base.as_ptr(), path.as_ptr(), $caseless),
                     $expected
                 );
             };
         }
-        t!("baz", false, false);
-        t!("baz", true, false);
-        t!("foo", false, true);
-        t!("foo", true, true);
-        t!("foo/bar", false, true);
-        t!("foo/bar", true, true);
-        t!("foo/BAR", true, true);
-        t!("FOO/BAR", true, true);
-        t!("FOO/bar", true, true);
-        t!("FOO", true, true);
+
+        t!(temp_dir.path(), "baz", false, false);
+        t!(temp_dir.path(), "baz", true, false);
+
+        t!(temp_dir.path(), "foo", false, true);
+        t!(temp_dir.path(), "foo", true, true);
+        t!(temp_dir.path(), "FOO", true, true);
+
+        t!(temp_dir.path(), "foo/bar", false, true);
```

Comment:
Does it make sense to add a test where the base dir is `temp_dir.path().join("Foo")` to test whether the base path is resolved case insensitively as well?

Reply:
The result will depend on the case sensitivity of the filesystem, so it can't be tested.